src/blueapi/core/context.py

```diff
@@ -51,12 +55,21 @@ class BlueskyContext:
     run_engine: RunEngine = field(
         default_factory=lambda: RunEngine(context_managers=[])
     )
+    plan_wrappers: Sequence[PlanWrapper] = field(default_factory=list)
     plans: Dict[str, Plan] = field(default_factory=dict)
     devices: Dict[str, Device] = field(default_factory=dict)
     plan_functions: Dict[str, PlanGenerator] = field(default_factory=dict)
 
     _reference_cache: Dict[Type, Type] = field(default_factory=dict)
 
+    def wrap(self, plan: MsgGenerator) -> MsgGenerator:
+        wrapped_plan = functools.reduce(
+            lambda wrapped, next_wrapper: next_wrapper(wrapped),
+            self.plan_wrappers,
+            plan,
```

Comment:
Might be obvious to people coming from python, but making plan be a kwarg `initializer=plan` would stop me running off to read the docs for functools every time I see it.

Reply:
Reduce explicitly disallows keyword arguments